tests/modules/hooks.c

```diff
@@ -287,9 +289,51 @@ void keyInfoCallback(RedisModuleCtx *ctx, RedisModuleEvent e, uint64_t sub, void
     }
 
     RedisModuleKeyInfoV1 *ei = data;
-    RedisModuleString *keyname = RedisModule_CreateStringPrintf(ctx, "key-info-%s", ei->key);
-    LogStringEvent(ctx, RedisModule_StringPtrLen(keyname, NULL), ei->key);
-    RedisModule_FreeString(ctx, keyname);
+    RedisModuleString *key = ei->key;
+    const char* keyname = RedisModule_StringPtrLen(key, NULL);
+    RedisModuleString *event_keyname = RedisModule_CreateStringPrintf(ctx, "key-info-%s", keyname);
+    LogStringEvent(ctx, RedisModule_StringPtrLen(event_keyname, NULL), keyname);
+    RedisModule_FreeString(ctx, event_keyname);
+
+    /* Open Key to check key exists */
+    RedisModuleKey *kp = RedisModule_OpenKey(ctx, key, REDISMODULE_READ);
+    if(kp != NULL) {
+        int nokey;
+        RedisModule_DictGetC(removed_event_log, (void*)keyname, strlen(keyname), &nokey);
```

Comment:
i don't think i understand the purpose of `nokey` (maybe because this is a superficial review), please add a comment to explain it.

Reply:
I removed `nokey` and use `RedisModule_DictReplaceC`.